sdk/kubeflow/trainer/api/trainer_client.py

```diff
@@ -170,29 +183,63 @@ def train(
         # Build the Trainer.
         trainer_crd = models.TrainerV1alpha1Trainer()
 
-        # Add number of nodes to the Trainer.
-        if trainer and trainer.num_nodes:
-            trainer_crd.num_nodes = trainer.num_nodes
+        if trainer:
+            # If users choose to use a custom training function.
+            if isinstance(trainer, types.CustomTrainer):
+                # Add number of nodes to the Trainer.
+                if trainer.num_nodes:
+                    trainer_crd.num_nodes = trainer.num_nodes
+
+                # Add resources per node to the Trainer.
+                if trainer.resources_per_node:
+                    trainer_crd.resources_per_node = utils.get_resources_per_node(
+                        trainer.resources_per_node
+                    )
 
-        # Add resources per node to the Trainer.
-        if trainer and trainer.resources_per_node:
-            trainer_crd.resources_per_node = utils.get_resources_per_node(
-                trainer.resources_per_node
-            )
+                # Add command and args to the Trainer.
+                trainer_crd.command = constants.DEFAULT_CUSTOM_COMMAND
+                # TODO: Support train function parameters.
+                trainer_crd.command, trainer_crd.args = (
+                    utils.get_entrypoint_using_train_func(
+                        runtime,
+                        trainer.func,
+                        trainer.func_args,
+                        trainer.pip_index_url,
+                        trainer.packages_to_install,
+                    )
+                )
 
-        # Add command and args to the Trainer if training function is set.
-        if trainer and trainer.func:
-
-            # TODO: Support train function parameters.
-            trainer_crd.command, trainer_crd.args = (
-                utils.get_entrypoint_using_train_func(
-                    runtime,
-                    trainer.func,
-                    trainer.func_args,
-                    trainer.pip_index_url,
-                    trainer.packages_to_install,
+            # If users choose to use a builtin trainer for post-training.
+            elif isinstance(trainer, types.BuiltinTrainer):
+                if trainer.config is None or not isinstance(
+                    trainer.config, types.TorchTuneConfig
+                ):
+                    raise ValueError(
+                        "The config is missing or not supported for the BuiltinTrainer. "
+                        "Currently, we only support `TorchTuneConfig` as parameter."
+                    )
```

Comment:
Let's make error in a way that, we don't need to always modify it when we support more trainers.
```suggestion
                        "The BuiltinTrainer config is invalid: {trainer.config}"
                    )
```